src/StructuredLogger/BinaryLogger/BinaryLogReplayEventSource.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.IO;
+using System.Text;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Shared;
+using System.Threading;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.Build.Logging.StructuredLogger
+{
+    /// <summary>
+    /// Interface for replaying a binary log file (*.binlog)
+    /// </summary>
+    internal interface IBinaryLogReplaySource :
+        IEventSource,
+        IBuildEventStringsReader,
+        IBuildFileReader,
+        IEmbeddedContentSource
+    { }
+
+    /// <summary>
+    /// Provides a method to read a binary log file (*.binlog) and replay all stored BuildEventArgs
+    /// by implementing IEventSource and raising corresponding events.
+    /// </summary>
+    /// <remarks>The class is public so that we can call it from MSBuild.exe when replaying a log file.</remarks>
+    public sealed class BinaryLogReplayEventSource : EventArgsDispatcher, IBinaryLogReplaySource

```

Comment:
What's your thinking for this type? Is this similar to the one in MSBuild? What should this type be used for? Only in the viewer repo?

Reply:
Oh, I guess this is temporary until we update to the shipped version of the MSBuild NuGet that has this? Is that right?

Reply:
Right. I made it `internal`